app/services/profit_sharing_service.py

```
@@ -558,12 +559,13 @@ async def setup_new_user_welcome_package(self, user_id: str) -> Dict[str, Any]:
                 
                 # Create welcome credit transaction
                 welcome_transaction = CreditTransaction(
-                    user_id=user_id,
+                    account_id=credit_account.id,
```

Ruff (0.12.2)

Undefined name `credit_account`

(F821)